scalalib/src/JavaModule.scala

```diff
@@ -585,39 +615,46 @@ trait JavaModule
 
   /**
    * Command to print the transitive dependency tree to STDOUT.
-   *
-   * @param inverse Invert the tree representation, so that the root is on the bottom.
-   * @param withCompile Include the compile-time only dependencies (`compileIvyDeps`, provided scope) into the tree.
-   * @param withRuntime Include the runtime dependencies (`runIvyDeps`, runtime scope) into the tree.
-   */
-  def ivyDepsTree(
-      inverse: Boolean = false,
-      withCompile: Boolean = false,
-      withRuntime: Boolean = false
-  ): Command[Unit] =
-    (withCompile, withRuntime) match {
-      case (true, true) =>
-        T.command {
-          printDepsTree(
-            inverse,
-            T.task {
-              transitiveCompileIvyDeps() ++ runIvyDeps()
-            }
-          )
-        }
-      case (true, false) =>
-        T.command {
-          printDepsTree(inverse, transitiveCompileIvyDeps)
-        }
-      case (false, true) =>
-        T.command {
-          printDepsTree(inverse, runIvyDeps)
-        }
-      case _ =>
-        T.command {
-          printDepsTree(inverse, T.task { Agg.empty[Dep] })
-        }
+   */
+  def ivyDepsTree(args: IvyDepsTreeArgs): Command[Unit] = {
+
+    val dependsOnModules = args.whatDependsOn.map(ModuleParser.javaOrScalaModule(_))
+
+    val (invalidModules, validModules) =
+      args.whatDependsOn.map(ModuleParser.javaOrScalaModule(_)).partitionMap(identity)
+
+    if (invalidModules.isEmpty) {
+      (args.withCompile, args.withRuntime) match {
```

Comment:
You could use `args.withCompile.value` which is a `Boolean` to match on `true` instead of `Flag(true)`, if you find it easier to read.

Reply:
I don't mind matching on the `Flag`.